gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -396,17 +419,19 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
               controller_type);
             impl_->controllers_.push_back(controller);
             load_params_from_yaml(
-              controller->get_lifecycle_node(),
+              controller->get_node(),
               impl_->param_file_,
               controller_name);
-            if (controller_name != "joint_state_controller") {
-              controller->get_lifecycle_node()->configure();
+            if (controller_name == "joint_state_controller") {
+              controller->assign_interfaces({}, std::move(loaned_state_ifs));
             }
+            controller->configure();
             start_controllers.push_back(controller_name);
           }
```

Comment:
All this bit is not needed here anymore in my opinion.

I believe these operations should be done only inside the controller manager through the standard interface (the load and switch services).